### PR TITLE
Split configuration into Next and Base config

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

- separate the config into NextConfig and BaseConfig (I'll come up with a better name when I figure out a better name for `auth0-session`) to make it easier to supply configuration that applies to the Next layer only (eg custom routes)
- Removed a bunch of docs and options that were only specific to the express SDK and got inherited by `auth0-session`

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -358,10 +276,91 @@ export interface AuthorizationParameters extends OidcAuthorizationParameters {
 }
 
 /**
- * See {@link Config}
+ * @ignore
+ */
+export type NextConfig = Pick<BaseConfig, 'routes' | 'identityClaimFilter'>;
```
Comment (adamjmcgrath) [inline src/config.ts:281]: Too add custom routes

```ts
export interface NextConfig extends Pick<BaseConfig, 'identityClaimFilter'> {
  routes: {
    postLogoutRedirect: string;
    callback: string;
    login: string;
    logout: string;
  };
}
```

File: src/auth0-session/get-config.ts [hunk 2]
```diff
@@ -133,8 +133,6 @@ const paramsSchema = Joi.object({
   issuerBaseURL: Joi.string().uri().required(),
   legacySameSiteCookie: Joi.boolean().optional().default(true),
   routes: Joi.object({
-    login: Joi.alternatives([Joi.string().uri({ relativeOnly: true }), Joi.boolean().valid(false)]).default('/login'),
-    logout: Joi.alternatives([Joi.string().uri({ relativeOnly: true }), Joi.boolean().valid(false)]).default('/logout'),
     callback: Joi.string().uri({ relativeOnly: true }).default('/callback'),
```
Comment (Widcket) [inline src/auth0-session/get-config.ts:136]: Why have a default callback URL?

**Review by Widcket — CHANGES_REQUESTED** [review]
Looks good, just have a question.

File: src/auth0-session/get-config.ts (hunk 2 above)
Comment (adamjmcgrath) [reply src/auth0-session/get-config.ts:136]: Good point 👍 - that was left over from the express code. Will remove that and make the field required.

File: src/auth0-session/get-config.ts [hunk 3]
```diff
@@ -133,9 +133,7 @@ const paramsSchema = Joi.object({
   issuerBaseURL: Joi.string().uri().required(),
   legacySameSiteCookie: Joi.boolean().optional().default(true),
   routes: Joi.object({
-    login: Joi.alternatives([Joi.string().uri({ relativeOnly: true }), Joi.boolean().valid(false)]).default('/login'),
-    logout: Joi.alternatives([Joi.string().uri({ relativeOnly: true }), Joi.boolean().valid(false)]).default('/logout'),
-    callback: Joi.string().uri({ relativeOnly: true }).default('/callback'),
+    callback: Joi.string().uri({ relativeOnly: true }).required(),
```
Comment (Widcket) [inline src/auth0-session/get-config.ts:136]: The login and logout URLs don't need to be required?

Comment (adamjmcgrath) [reply src/auth0-session/get-config.ts:136]: I've removed the login and logout URLs from the config as they are not used by `auth0-session`

**Review by Widcket — APPROVED** [review]

**Pull request merged**